Lib/test/test_mmap.py

```diff
@@ -264,6 +264,111 @@ def test_bad_file_desc(self):
         # Try opening a bad file descriptor...
         self.assertRaises(OSError, mmap.mmap, -2, 4096)
 
+    def test_unexpected_mmap_close_scenario_1(self):
+        # See gh-103987
+        with open(TESTFN, "wb+") as f:
+            data = b'aabaac\x00deef\x00\x00aa\x00'
+            n = len(data)
+            f.write(data)
+            f.flush()
+
+            class X:
+                def __index__(self):
+                    m.close()
+                    return 1
+
+            m = mmap.mmap(f.fileno(), n, access=mmap.ACCESS_READ)
+
+        self.assertEqual(m[1], 97)
+        with self.assertRaises(ValueError):
+            m[X()] # should not crash
+
+
+    def test_unexpected_mmap_close_scenario_2(self):
+        # See gh-103987
+        with open(TESTFN, "wb+") as f:
+            data = b'aabaac\x00deef\x00\x00aa\x00'
+            n = len(data)
+            f.write(data)
+            f.flush()
+
+            class X:
+                def __index__(self):
+                    m.close()
+                    return 1
+
+            m = mmap.mmap(f.fileno(), n, access=mmap.ACCESS_READ)
+
+        self.assertEqual(m[1], 97)
+        with self.assertRaises(ValueError):
+            m[X():5] # should not crash
+
+    def test_unexpected_mmap_close_scenario_3(self):
+        # See gh-103987
+        with open(TESTFN, "wb+") as f:
+            data = b'aabaac\x00deef\x00\x00aa\x00'
+            n = len(data)
+            f.write(data)
+            f.flush()
+
+            m = mmap.mmap(f.fileno(), n, access=mmap.ACCESS_READ)
+
+        m.close()
+        with self.assertRaises(ValueError):
+            m["abc"]
+
+    def test_unexpected_mmap_close_scenario_4(self):
+        # See gh-103987
+        with open(TESTFN, "wb+") as f:
+            data = b'aabaac\x00deef\x00\x00aa\x00'
+            n = len(data)
+            f.write(data)
+            f.flush()
+
+            m = mmap.mmap(f.fileno(), n, access=mmap.ACCESS_READ)
+
+        m.close()
+        with self.assertRaises(ValueError):
+            m[0:5] # should not crash
```

Comment:
This is not **unexpected** mmap close. Remove this test, or move to other places.